datasafe-business/src/test/java/de/adorsys/datasafe/business/impl/e2e/BasicFunctionalityWithConcurrencyTest.java

```diff
@@ -237,17 +209,58 @@ private void readOriginUserInboxAndWriteToTargetUserPrivate(UserIDAuth originUse
         countDownLatch.countDown();
     }
 
-    @AfterAll
-    @SneakyThrows
-    static void exportReport() {
-        loadReport.stream().sorted().forEach(log::info);
+    private static void generateTestFile(String testFilePath, int testFileSizeInBytes) {
+        RandomAccessFile originTestFile = null;
+        try {
+            originTestFile = new RandomAccessFile(testFilePath, "rw");
+            MappedByteBuffer out = originTestFile.getChannel()
+                    .map(FileChannel.MapMode.READ_WRITE, 0, testFileSizeInBytes);
+
+            for (int i = 0; i < testFileSizeInBytes; i++) {
+                out.put((byte) 'x');
+            }
+        } catch (IOException e) {
+            log.error(e.getMessage());
+        }
+
     }
 
-    private static byte[] generateTestData(int testFileSizeInBytes) {
-        byte[] data = new byte[testFileSizeInBytes];
+    @ValueSource
+    protected static Stream<Arguments> differentThreadsTestOptions() {
+        Stream<StorageDescriptor> storageDescriptorMap = storages();
+        List<Arguments> arguments = new ArrayList<>();
+
+        storageDescriptorMap.forEach(storageDescriptor -> {
+            //30kb - 4 threads pool size
+            arguments.add(Arguments.of(storageDescriptor, 1024 * 30, 4));
+            //30kb - 8 threads pool size
+            arguments.add(Arguments.of(storageDescriptor, 1024 * 30, 8));
+            //60kb - 4 threads pool size
+            arguments.add(Arguments.of(storageDescriptor, 1024 * 60, 4));
+            //60kb - 8 threads pool size
+            arguments.add(Arguments.of(storageDescriptor, 1024 * 60, 8));
+            //5Mb - 4 threads pool size
+            arguments.add(Arguments.of(storageDescriptor, 1024 * 1024 * 5, 4));
+            //5Mb - 8 threads pool size
+            arguments.add(Arguments.of(storageDescriptor, 1024 * 1024 * 5, 8));
+        });
+        return arguments.stream();
+    }
 
-        IntStream.range(0, testFileSizeInBytes).forEach(it -> data[it] = 'x');
+    @AfterAll
+    public static void cleanUpTestFile() throws IOException {
+        Files.delete(Paths.get("./test.txt"));
```

Comment:
It is really better to use Junit TempDir annotation for temporary resource management